app/Models/Feature/Feature.php

```diff
@@ -112,6 +120,14 @@ public function altTypes()
         return $this->hasMany('App\Models\Feature\Feature', 'parent_id');
     }
 
+        /**
+     * Get alternate types of this feature, but only those visible.
+     */
+    public function altTypesVisible()
+    {
+        return $this->hasMany('App\Models\Feature\Feature', 'parent_id')->where('is_visible', 1);
```

Comment:
Likewise, this can probably just be removed, especially since it's only used in the one place...

Reply:
Removed, as no longer necessary with changes!